File: Back/OnlineJudge/src/main/java/com/oj/onlinejudge/domain/entity/Announcement.java
package com.oj.onlinejudge.domain.entity;

// 公告实体：映射公告表 announcement

import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import io.swagger.v3.oas.annotations.media.Schema;

import java.time.LocalDateTime;

import lombok.Data;

@Data
@TableName("announcement")
@Schema(name = "Announcement", description = "公告实体")
public class Announcement {
    @TableId
    @Schema(description = "公告ID")
    private Long id;
    @Schema(description = "标题")
    private String title;
    @Schema(description = "内容")
    private String content;
    @TableField("time")
    @Schema(description = "发布时间")
    private LocalDateTime time;
    @TableField("is_pinned")
    @Schema(description = "是否置顶")
    private Boolean isPinned;
    @TableField("updated_at")
    @Schema(description = "更新时间")
    private LocalDateTime updatedAt;
    @TableField("is_active")
    @Schema(description = "是否启用")
    private Boolean isActive;
}

File: Front/OnlineJudge/oj-vue/src/types/dto/announcement.ts
export interface AnnouncementVO {
  id: string;
  title: string;
  content: string;
  time: string;
  isPinned: boolean;
  isActive: boolean;
  createdAt?: string;
  updatedAt?: string;
}

export interface AnnouncementRequest {
  title: string;
  content: string;
  time?: string;
  isPinned?: boolean;
  isActive?: boolean;
}

export interface AnnouncementQuery {
  page?: number;
  size?: number;
  keyword?: string;
  pinnedOnly?: boolean;
  isPinned?: boolean;
}
